### PR TITLE
implemented network acl nuke

### PR DESCRIPTION
**Pull request by hongil0316**
## Description

Partially Fixing https://github.com/gruntwork-io/cloud-nuke/issues/476

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: aws/resources/ec2_network_acl.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+				},
+			},
+		},
+	)
+	if err != nil {
+		logging.Debugf("[Network ACL] Failed to describe network ACL: %s", err)
+		return err
+	}
+
+	if len(networkACLs.NetworkAcls) <= 0 {
```
Comment (denis256) [inline aws/resources/ec2_network_acl.go:177]: I suspect the slice length can't be bellow 0

File: aws/resources/ec2_network_acl.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+func (nacl *NetworkACL) nukeAssociatedSubnets(id *string) error {
+	resp, err := nacl.Client.DescribeNetworkAcls(&ec2.DescribeNetworkAclsInput{
+		NetworkAclIds: []*string{id},
+	})
+	if err != nil {
+		logging.Debugf("[Network ACL] Failed to describe network ACL: %s", err)
+		return err
+	}
+
+	if len(resp.NetworkAcls) <= 0 {
```
Comment (denis256) [inline aws/resources/ec2_network_acl.go:146]: I suspect the slice length can't be bellow 0

File: aws/resources/ec2_network_acl.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+			NetworkAclId:  defaultNetworkAclID,
+		})
+		if err != nil {
+			logging.Debug(fmt.Sprintf("Failed to replace network ACL association: %s to default",
+				aws.StringValue(association.NetworkAclAssociationId)))
+			return errors.WithStackTrace(err)
+		}
+		logging.Debug(fmt.Sprintf("Successfully replaced network ACL association: %s to default",
+			aws.StringValue(association.NetworkAclAssociationId)))
+
```
Comment (denis256) [inline aws/resources/ec2_network_acl.go:199]: unnecessary trailing newline

File: aws/resources/ec2_network_acl_test.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+			configObj: config.ResourceType{
+				ExcludeRule: config.FilterRule{
+					TimeAfter: aws.Time(now),
+				}},
+			expected: []string{
+				testId1,
+			},
+		},
+	}
+	for name, tc := range tests {
```
Comment (denis256) [inline aws/resources/ec2_network_acl_test.go:123]: tc is not reinitialized, in test case, I suspect all tests will run with the same tc instance

File: aws/resources/ec2_network_acl.go (hunk 1 above)
Comment (jameskwon316) [reply aws/resources/ec2_network_acl.go:177]: Will remove.

File: aws/resources/ec2_network_acl.go (hunk 2 above)
Comment (jameskwon316) [reply aws/resources/ec2_network_acl.go:146]: Will remove.

File: aws/resources/ec2_network_acl.go (hunk 3 above)
Comment (jameskwon316) [reply aws/resources/ec2_network_acl.go:199]: Will remove.

File: aws/resources/ec2_network_acl_test.go (hunk 4 above)
Comment (jameskwon316) [reply aws/resources/ec2_network_acl_test.go:123]: Not really sure about this comment. Can you elaborate and what you are suggesting?

Comment (hongil0316) [reply aws/resources/ec2_network_acl_test.go:123]: I wonder if we should revisit this at a follow-up PR. I can look into whether removing `t.Parallel()` fixes the problem and explore other option if necessary.

**Review by denis256 — APPROVED** [review]

**Pull request merged**